eth/tracers/api.go

```diff
@@ -725,139 +733,219 @@ txloop:
+	msg, _ := core.TransactionToMessage(tx, signer, block.BaseFee())
+	txctx := &Context{
+		BlockHash:   block.Hash(),
+		BlockNumber: block.Number(),
+		TxIndex:     txIdx,
+		TxHash:      tx.Hash(),
 	}
-	statedb, release, err := api.backend.StateAtBlock(ctx, parent, reexec, nil, true, false)
+	_, blockCtx, statedb, release, err := api.backend.StateAtTransaction(ctx, block, txIdx, defaultTraceReexec)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 	defer release()
-	// Retrieve the tracing configurations, or use default values
-	var (
-		logConfig logger.Config
-		txHash    common.Hash
-	)
+
+	var dump *os.File
+
+	dump, err = createTraceDumpFile(block.Hash(), tx.Hash(), txIdx, canon)
+	if err != nil {
+		return "", err
+	}
+
+	// Swap out the noop logger to the standard traces
+	writer := bufio.NewWriter(dump)
+
+	var logConfig *logger.Config
 	if config != nil {
-		logConfig = config.Config
-		txHash = config.TxHash
+		logConfig = config
 	}
 
-	// Execute transaction, either tracing all or just the requested one
-	var (
-		dumps       []string
-		signer      = types.MakeSigner(api.backend.ChainConfig(), block.Number(), block.Time())
-		chainConfig = api.backend.ChainConfig()
-		vmctx       = core.NewEVMBlockContext(block.Header(), api.chainContext(ctx), nil)
-		canon       = true
-	)
-	// Check if there are any overrides: the caller may wish to enable a future
-	// fork when executing this block. Note, such overrides are only applicable to the
-	// actual specified block, not any preceding blocks that we have to go through
-	// in order to obtain the state.
-	// Therefore, it's perfectly valid to specify `"futureForkBlock": 0`, to enable `futureFork`
-	if config != nil && config.Overrides != nil {
-		// Note: This copies the config, to not screw up the main config
-		chainConfig, canon = overrideConfig(chainConfig, config.Overrides)
+	logger := logger.NewJSONLogger(logConfig, writer)
+	tracer := Tracer{
+		Hooks:     logger,
+		Stop:      func(err error) {},
+		GetResult: func() (json.RawMessage, error) { return nil, nil },
+	}
+	traceTimeout := new(time.Duration)
+	*traceTimeout = 1 * time.Hour
+	_, err = api.execTx(ctx, tx, msg, txctx, blockCtx, statedb, &tracer, nil, *traceTimeout)
+	if err != nil {
+		return "", err
 	}
+	if err = writer.Flush(); err != nil {
+		return "", err
+	}
+	dump.Close()
 
-	evm := vm.NewEVM(vmctx, statedb, chainConfig, vm.Config{})
-	if beaconRoot := block.BeaconRoot(); beaconRoot != nil {
-		core.ProcessBeaconBlockRoot(*beaconRoot, evm)
+	return dump.Name(), nil
+}
+
+func traceCallExecOpt(config *TraceCallConfig) (*traceExecOptions, error) {
+	if config == nil {
+		return traceExecOpt(false, nil)
 	}
-	if chainConfig.IsPrague(block.Number(), block.Time()) {
-		core.ProcessParentBlockHash(block.ParentHash(), evm)
+	return traceExecOpt(false, &config.TraceConfig)
+}
+
+type traceExecOptions struct {
+	txTimeout time.Duration
+	reexec    uint64
+	parallel  bool
+}
```

Comment:
`traceBlock` and `traceTx` (now `execTx`) are now agnostic to the type of tracer (call vs standard-trace vs js/native-trace).  Now, we extract the trace execution parameters from the config type in the outer method call (`StdTrace`, `TraceConfig`, `TraceCallConfig`) into a `traceExecOptions` instance which is used by the internal tracing methods.

This reduces the size of some method signatures and prevents duplication of config instantiation logic.